atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/DefaultDeleteExecutor.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl;
+
+import com.google.common.collect.Multimaps;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.logging.LoggingArgs;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.UnsafeArg;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import com.palantir.util.RateLimitedLogger;
+import java.time.Duration;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.RejectedExecutionException;
+
+public final class DefaultDeleteExecutor implements DeleteExecutor {
+    private static final SafeLogger log = SafeLoggerFactory.get(DefaultDeleteExecutor.class);
+    private static final RateLimitedLogger deleteExecutorRateLimitedLogger = new RateLimitedLogger(log, 1.0);
+    private static final String ERROR_MSG =
+            "This isn't a bug but it should be infrequent if all nodes of your KV service are"
+                    + " running. Delete has stronger consistency semantics than read/write and must talk to all nodes"
+                    + " instead of just talking to a quorum of nodes. "
+                    + "Failed to delete keys for table: {} from an uncommitted transaction; "
+                    + " sweep should eventually clean these values.";
+
+    private final KeyValueService keyValueService;
+    private final ExecutorService executorService;
+
+    public DefaultDeleteExecutor(KeyValueService keyValueService, ExecutorService executorService) {
```

Comment:
I think that's probably generally assumed? e.g. Cassandra KVS closes out a bunch of these, and I think it gets them in constructors. I'm happy to add if you'd like, but feels pretty standard to me.

Reply:
OK, yeah seems fine. I think I was thinking of the `LockServiceImpl` which does have specific behaviours for this, but that's mostly because of legacy